--- /psquare/developer/ANDROID/CYANOGENMOD7/frameworks/base/include/ui/InputManager.h
+++ /psquare/developer/ANDROID/CYANOGENMOD/frameworks/base/include/ui/InputManager.h
@@ -40,6 +40,9 @@
 class InputDispatcherInterface;
 class InputDispatcherPolicyInterface;
 class InputDispatcherThread;
+
+class InputRemote;
+class InputRemoteThread;
 
 /*
  * The input manager is the core of the system event processing.
@@ -107,6 +110,9 @@
     sp<InputDispatcherInterface> mDispatcher;
     sp<InputDispatcherThread> mDispatcherThread;
 
+	sp<InputRemote> mRemote;
+    sp<InputRemoteThread> mRemoteThread;
+
     void initialize();
 };
 
