--- /psquare/developer/ANDROID/CYANOGENMOD7/frameworks/base/include/ui/InputDispatcher.h
+++ /psquare/developer/ANDROID/CYANOGENMOD/frameworks/base/include/ui/InputDispatcher.h
@@ -325,6 +325,9 @@
      */
     virtual bool checkInjectEventsPermissionNonReentrant(
             int32_t injectorPid, int32_t injectorUid) = 0;
+
+    /* Handle mouse cursor motion before queueing to skip throttling and other undesired operations. */
+    virtual void handleCursorMotion(int32_t x, int32_t y) { }
 };
 
 
