--- /psquare/developer/ANDROID/CYANOGENMOD7/frameworks/base/core/jni/AndroidRuntime.cpp
+++ /psquare/developer/ANDROID/CYANOGENMOD/frameworks/base/core/jni/AndroidRuntime.cpp
@@ -118,6 +118,7 @@
 extern int register_android_graphics_Xfermode(JNIEnv* env);
 extern int register_android_graphics_PixelFormat(JNIEnv* env);
 extern int register_com_android_internal_graphics_NativeUtils(JNIEnv *env);
+extern int register_com_android_internal_graphics_OSD2Cursor(JNIEnv *env);
 extern int register_android_view_Display(JNIEnv* env);
 extern int register_android_view_Surface(JNIEnv* env);
 extern int register_android_view_ViewRoot(JNIEnv* env);
@@ -1247,6 +1248,7 @@
     REG_JNI(register_android_graphics_Xfermode),
     REG_JNI(register_android_graphics_YuvImage),
     REG_JNI(register_com_android_internal_graphics_NativeUtils),
+    REG_JNI(register_com_android_internal_graphics_OSD2Cursor),
 
     REG_JNI(register_android_database_CursorWindow),
     REG_JNI(register_android_database_SQLiteCompiledSql),
