--- mp4enc_api.h.original	2012-02-09 22:22:07.000000000 +0100
+++ mp4enc_api.h.psquare	2012-02-09 15:37:04.000000000 +0100
@@ -446,6 +446,9 @@
     *   @return true for correct operation; false if error happens
     */
     OSCL_IMPORT_REF Bool    PVCleanUpVideoEncoder(VideoEncControls *encCtrl);
+    OSCL_IMPORT_REF void ConvertYUV420SemiPlanarToYUV420Planar(
+        uint8_t *inyuv, uint8_t* outyuv,
+        int32_t width, int32_t height);
 
 #ifdef __cplusplus
 }
