--- Android.mk.original	2012-02-09 22:22:07.000000000 +0100
+++ Android.mk.psquare	2012-02-09 15:45:56.000000000 +0100
@@ -6,10 +6,8 @@
     src/bitstream_io.cpp \
     src/combined_encode.cpp \
     src/datapart_encode.cpp \
-    src/dct.cpp \
     src/findhalfpel.cpp \
     src/fastcodemb.cpp \
-    src/fastidct.cpp \
     src/fastquant.cpp \
     src/me_utils.cpp \
     src/mp4enc_api.cpp \
@@ -17,16 +15,28 @@
     src/motion_est.cpp \
     src/motion_comp.cpp \
     src/sad.cpp \
-    src/sad_halfpel.cpp \
     src/vlc_encode.cpp \
     src/vop.cpp
 
 
+ifeq ($(ARCH_ARM_HAVE_NEON),true)
+LOCAL_SRC_FILES += src/dct_neon.s src/sad_neon.s src/sad_halfpel_neon.s src/fastidct_neon.s src/dct4sad.c ColorConverter_neon.s
+else
+LOCAL_SRC_FILES += src/dct.cpp src/fastidct.cpp src/sad_halfpel.cpp
+endif
+
 LOCAL_MODULE := libstagefright_m4vh263enc
 
 LOCAL_CFLAGS := \
     -DBX_RC \
     -DOSCL_IMPORT_REF= -DOSCL_UNUSED_ARG= -DOSCL_EXPORT_REF=
+ifeq ($(ARCH_ARM_HAVE_NEON),true)
+LOCAL_CFLAGS += -DASM_OPT
+endif
+
+ifeq ($(ENABLE_VIDEOENCODER_ROTATE),true)
+LOCAL_CPPFLAGS += -DAML_VIDEOENCODER_ROTATE
+endif
 
 LOCAL_C_INCLUDES := \
     $(LOCAL_PATH)/src \
