--- /psquare/developer/ANDROID/CYANOGENMOD7/frameworks/base/core/java/android/view/MotionEvent.java
+++ /psquare/developer/ANDROID/CYANOGENMOD/frameworks/base/core/java/android/view/MotionEvent.java
@@ -781,6 +781,16 @@
         return mDataSamples[mLastDataSampleIndex + SAMPLE_Y] + mYOffset;
     }
 
+    /** @hide */
+    public final float getXOffset() {
+        return mXOffset;
+    }
+
+    /** @hide */
+    public final float getYOffset() {
+        return mYOffset;
+    }
+
     /**
      * {@link #getPressure(int)} for the first pointer index (may be an
      * arbitrary pointer identifier).
