--- /psquare/developer/ANDROID/CYANOGENMOD7/frameworks/base/core/java/android/view/ViewRoot.java
+++ /psquare/developer/ANDROID/CYANOGENMOD/frameworks/base/core/java/android/view/ViewRoot.java
@@ -112,6 +112,8 @@
     
     private static int sDrawTime;
 
+    private boolean mMouseDown = false;
+
     long mLastTrackballTime = 0;
     final TrackballAxis mTrackballAxisX = new TrackballAxis();
     final TrackballAxis mTrackballAxisY = new TrackballAxis();
@@ -216,6 +218,8 @@
 
     final ViewConfiguration mViewConfiguration;
 
+    static IWindowManager wm;
+
     /**
      * see {@link #playSoundEffect(int)}
      */
@@ -228,9 +232,8 @@
             if (!mInitialized) {
                 try {
                     InputMethodManager imm = InputMethodManager.getInstance(mainLooper);
-                    sWindowSession = IWindowManager.Stub.asInterface(
-                            ServiceManager.getService("window"))
-                            .openSession(imm.getClient(), imm.getInputContext());
+                    wm = IWindowManager.Stub.asInterface(ServiceManager.getService("window"));
+                    sWindowSession = wm.openSession(imm.getClient(), imm.getInputContext());
                     mInitialized = true;
                 } catch (RemoteException e) {
                 }
@@ -2873,7 +2876,18 @@
 
     private void dispatchMotion(MotionEvent event, boolean sendDone) {
         int source = event.getSource();
-        if ((source & InputDevice.SOURCE_CLASS_POINTER) != 0) {
+        if (((source & InputDevice.SOURCE_MOUSE) ^ InputDevice.SOURCE_MOUSE) == 0) {
+            int action = event.getActionMasked();
+            if (action == MotionEvent.ACTION_DOWN)
+                mMouseDown = true;
+            if (action == MotionEvent.ACTION_UP)
+                mMouseDown = false;
+
+            if (mMouseDown || action == MotionEvent.ACTION_UP)
+                dispatchPointer(event, sendDone);
+            else if (sendDone)
+                finishInputEvent();
+        } else if ((source & InputDevice.SOURCE_CLASS_POINTER) != 0) {
             dispatchPointer(event, sendDone);
         } else if ((source & InputDevice.SOURCE_CLASS_TRACKBALL) != 0) {
             dispatchTrackball(event, sendDone);
