--- /psquare/developer/ANDROID/CYANOGENMOD7/frameworks/base/core/java/android/view/Window.java
+++ /psquare/developer/ANDROID/CYANOGENMOD/frameworks/base/core/java/android/view/Window.java
@@ -480,6 +480,14 @@
         public Display getDefaultDisplay() {
             return mDefaultDisplay;
         }
+
+        public int getMouseCursorType() {
+            return mWindowManager.getMouseCursorType();
+        }
+
+        public int setMouseCursorType(int type) {
+            return mWindowManager.setMouseCursorType(type);
+        }
         
         private final WindowManager mWindowManager;
 
