--- /psquare/developer/ANDROID/CYANOGENMOD7/frameworks/base/core/java/android/view/WindowManager.java
+++ /psquare/developer/ANDROID/CYANOGENMOD/frameworks/base/core/java/android/view/WindowManager.java
@@ -64,6 +64,25 @@
      * @param view The view to be removed.
      */
     public void removeViewImmediate(View view);
+
+    /** No mouse cursor displayed. */
+    public static final int MOUSE_CURSOR_NONE = 0;
+    /** Mouse cursor using a Surface. */
+    public static final int MOUSE_CURSOR_SURFACE = 1;
+    /** Mouse cursor on OSD2 */
+    public static final int MOUSE_CURSOR_OSD2 = 2;
+
+    /**
+     * Get the mouse cursor type.
+     * @return mouse cursor type or -1 for error
+     */
+    public int getMouseCursorType();
+
+    /**
+     * Set the mouse cursor type.
+     * @return mouse cursor type
+     */
+    public int setMouseCursorType(int type);
     
     public static class LayoutParams extends ViewGroup.LayoutParams
             implements Parcelable {
@@ -1054,6 +1073,14 @@
                 gravity = o.gravity;
                 changes |= LAYOUT_CHANGED;
             }
+            if (horizontalMargin != o.horizontalMargin) {
+                horizontalMargin = o.horizontalMargin;
+                changes |= LAYOUT_CHANGED;
+            }
+            if (verticalMargin != o.verticalMargin) {
+                verticalMargin = o.verticalMargin;
+                changes |= LAYOUT_CHANGED;
+            }
             if (format != o.format) {
                 format = o.format;
                 changes |= FORMAT_CHANGED;
