--- /psquare/developer/ANDROID/CYANOGENMOD7/frameworks/base/core/java/android/view/WindowManagerImpl.java
+++ /psquare/developer/ANDROID/CYANOGENMOD/frameworks/base/core/java/android/view/WindowManagerImpl.java
@@ -18,6 +18,8 @@
 
 import android.graphics.PixelFormat;
 import android.os.IBinder;
+import android.os.RemoteException;
+import android.os.ServiceManager;
 import android.util.AndroidRuntimeException;
 import android.util.Config;
 import android.util.Log;
@@ -325,7 +327,23 @@
     public Display getDefaultDisplay() {
         return new Display(Display.DEFAULT_DISPLAY);
     }
-
+    public int getMouseCursorType() {
+        IWindowManager wm = IWindowManager.Stub.asInterface(ServiceManager.getService("window"));
+        try {
+            return wm.getMouseCursorType();
+        } catch (RemoteException e) {
+        }
+        return -1;
+    }
+
+    public int setMouseCursorType(int type) {
+        IWindowManager wm = IWindowManager.Stub.asInterface(ServiceManager.getService("window"));
+        try {
+            return wm.setMouseCursorType(type);
+        } catch (RemoteException e) {
+        }
+        return -1;
+    }
     private View[] mViews;
     private ViewRoot[] mRoots;
     private WindowManager.LayoutParams[] mParams;
