--- InputManager.java.original	2012-02-23 18:57:45.000000000 +0100
+++ InputManager.java.psquare	2012-02-29 02:28:51.000000000 +0100
@@ -79,6 +79,7 @@
     private static native InputDevice nativeGetInputDevice(int deviceId);
     private static native void nativeGetInputConfiguration(Configuration configuration);
     private static native int[] nativeGetInputDeviceIds();
+	private static native void nativeSetTvOutStatus(boolean on);
     private static native String nativeDump();
     
     // Input event injection constants defined in InputDispatcher.h.
@@ -326,6 +327,10 @@
         nativeSetInputDispatchMode(enabled, frozen);
     }
     
+	public void setTvOutStatus(boolean on) {
+        nativeSetTvOutStatus(on);
+    }
+		
     public void dump(PrintWriter pw) {
         String dumpStr = nativeDump();
         if (dumpStr != null) {
@@ -539,5 +544,10 @@
             }
             return result;
         }
+
+//        @SuppressWarnings("unused")
+//        public void handleCursorMotion(int x, int y) {
+//            mWindowManagerService.mInputMonitor.handleCursorMotion(x, y);
+//        }
     }
 }
