--- /psquare/developer/ANDROID/CYANOGENMOD7/frameworks/base/core/java/android/view/SurfaceView.java
+++ /psquare/developer/ANDROID/CYANOGENMOD/frameworks/base/core/java/android/view/SurfaceView.java
@@ -726,6 +726,7 @@
             switch (type) {
             case SURFACE_TYPE_NORMAL:
             case SURFACE_TYPE_PUSH_BUFFERS:
+            case SURFACE_TYPE_NO_BLEND:
                 mRequestedType = type;
                 if (mWindow != null) {
                     updateWindow(false, false);
