--- /psquare/developer/ANDROID/CYANOGENMOD7/frameworks/base/libs/ui/InputDispatcher.cpp
+++ /psquare/developer/ANDROID/CYANOGENMOD/frameworks/base/libs/ui/InputDispatcher.cpp
@@ -2156,6 +2156,9 @@
     }
     mPolicy->interceptGenericBeforeQueueing(eventTime, /*byref*/ policyFlags);
 
+    if ((source & AINPUT_SOURCE_MOUSE) == AINPUT_SOURCE_MOUSE && action == AMOTION_EVENT_ACTION_MOVE)
+        mPolicy->handleCursorMotion((int32_t)pointerCoords[0].x, (int32_t)pointerCoords[0].y);
+
     bool needWake;
     { // acquire lock
         AutoMutex _l(mLock);
