--- /psquare/developer/ANDROID/CYANOGENMOD/frameworks/base/include/ui/InputReader.h.2
+++ /psquare/developer/ANDROID/CYANOGENMOD/frameworks/base/include/ui/InputReader.h
@@ -736,6 +736,15 @@
 
     // Immutable calibration parameters in parsed form.
     struct Calibration {
+        // Pointercal
+        float pointercalX1;
+        float pointercalX2;
+        float pointercalX3;
+        float pointercalY1;
+        float pointercalY2;
+        float pointercalY3;
+        float pointercalScale;
+
         // Touch Size
         enum TouchSizeCalibration {
             TOUCH_SIZE_CALIBRATION_DEFAULT,
