--- alsactl.1.original	2012-02-01 21:20:31.000000000 +0100
+++ alsactl.1.psquare	2012-02-01 22:34:29.000000000 +0100
@@ -43,7 +43,7 @@
 
 .TP
 \fI\-f, \-\-file\fP
-Select the configuration file to use. The default is /var/lib/alsa/asound.state.
+Select the configuration file to use. The default is /etc/asound.state.
 
 .TP
 \fI\-F, \-\-force\fP
@@ -90,7 +90,7 @@
 is used.
 
 .SH FILES
-\fI/var/lib/alsa/asound.state\fP (or whatever file you specify with the
+\fI/etc/asound.state\fP (or whatever file you specify with the
 \fB\-f\fP flag) is used to store current settings for your
 soundcards. The settings include all the usual soundcard mixer
 settings.  More importantly, alsactl is
