--- /psquare/developer/ANDROID/CYANOGENMOD7/frameworks/base/include/surfaceflinger/ISurfaceComposer.h
+++ /psquare/developer/ANDROID/CYANOGENMOD/frameworks/base/include/surfaceflinger/ISurfaceComposer.h
@@ -43,6 +43,7 @@
         eSecure             = 0x00000080,
         eNonPremultiplied   = 0x00000100,
         ePushBuffers        = 0x00000200,
+        eNoBlend            = 0x00000400,
 
         eFXSurfaceNormal    = 0x00000000,
         eFXSurfaceBlur      = 0x00010000,
@@ -66,7 +67,8 @@
         eLayerFrozen        = 0x02,
         eLayerDither        = 0x04,
         eLayerFilter        = 0x08,
-        eLayerBlurFreeze    = 0x10
+        eLayerBlurFreeze    = 0x10,
+        eLayerNoBlend       = 0x20
     };
 
     enum {
