--- /psquare/developer/ANDROID/CYANOGENMOD7/frameworks/base/core/java/android/database/AbstractCursor.java
+++ /psquare/developer/ANDROID/CYANOGENMOD/frameworks/base/core/java/android/database/AbstractCursor.java
@@ -204,7 +204,7 @@
      * @param window
      */
     public void fillWindow(int position, CursorWindow window) {
-        if (position < 0 || position >= getCount()) {
+        if (position < 0 || position > getCount()) {
             return;
         }
         window.acquireReference();
