--- /psquare/developer/ANDROID/CYANOGENMOD/frameworks/base/services/jni/com_android_server_InputManager.cpp.2
+++ /psquare/developer/ANDROID/CYANOGENMOD/frameworks/base/services/jni/com_android_server_InputManager.cpp
@@ -27,7 +27,9 @@
 #include "JNIHelp.h"
 #include "jni.h"
 #include <limits.h>
+#include <stdlib.h>
 #include <android_runtime/AndroidRuntime.h>
+#include <cutils/properties.h>
 #include <ui/InputReader.h>
 #include <ui/InputDispatcher.h>
 #include <ui/InputManager.h>
@@ -39,6 +41,10 @@
 #include "../../core/jni/android_view_InputChannel.h"
 #include "com_android_server_PowerManagerService.h"
 #include <cutils/properties.h>
+
+namespace {
+    int32_t hwrotation;
+}
 
 namespace android {
 
@@ -278,6 +284,10 @@
     mDisplayWidth(-1), mDisplayHeight(-1), mDisplayOrientation(ROTATION_0) {
     JNIEnv* env = jniEnv();
 
+    char propBuf[PROPERTY_VALUE_MAX];
+    property_get("input.hwrotation", propBuf, "0");
+    hwrotation = (atoi(propBuf) / 90) % 4;
+
     mCallbacksObj = env->NewGlobalRef(callbacksObj);
 
     sp<EventHub> eventHub = new EventHub();
@@ -426,13 +436,13 @@
 
         if (mDisplayWidth > 0) {
             if (width) {
-                *width = mDisplayWidth;
+                *width = (hwrotation % 2) ? mDisplayHeight : mDisplayWidth;
             }
             if (height) {
-                *height = mDisplayHeight;
+                *height = (hwrotation % 2) ? mDisplayWidth : mDisplayHeight;
             }
             if (orientation) {
-                *orientation = mDisplayOrientation;
+                *orientation = (mDisplayOrientation + hwrotation) % 4;
             }
             result = true;
         }
