--- /psquare/developer/ANDROID/CYANOGENMOD7/frameworks/base/libs/surfaceflinger_client/SurfaceComposerClient.cpp
+++ /psquare/developer/ANDROID/CYANOGENMOD/frameworks/base/libs/surfaceflinger_client/SurfaceComposerClient.cpp
@@ -487,6 +487,13 @@
 {
     layer_state_t* s = lockLayerState(id);
     if (!s) return BAD_INDEX;
+
+    if (mask == ISurfaceComposer::eNoBlend) {
+        mask = ISurfaceComposer::eLayerNoBlend;
+        if (flags == ISurfaceComposer::eNoBlend)
+            flags = ISurfaceComposer::eLayerNoBlend;
+    }
+
     s->what |= ISurfaceComposer::eVisibilityChanged;
     s->flags &= ~mask;
     s->flags |= (flags & mask);
