--- /psquare/developer/ANDROID/CYANOGENMOD7/frameworks/base/core/java/android/content/res/Configuration.java
+++ /psquare/developer/ANDROID/CYANOGENMOD/frameworks/base/core/java/android/content/res/Configuration.java
@@ -311,7 +311,7 @@
      * Set this object to the system defaults.
      */
     public void setToDefaults() {
-        fontScale = 0;
+        fontScale = 1;
         mcc = mnc = 0;
         locale = null;
         userSetLocale = false;
