--- wifi.c.original	2012-03-05 05:22:50.542898416 +0100
+++ wifi.c.psquare	2012-03-05 07:07:58.935466732 +0100
@@ -222,6 +222,7 @@
 
 static int insmod(const char *filename, const char *args)
 {
+#if 0
     void *module;
     unsigned int size;
     int ret;
@@ -240,10 +241,13 @@
     free(module);
 
     return ret;
+#endif
+	return 0;
 }
 
 static int rmmod(const char *modname)
 {
+#if 0
     int ret = -1;
     int maxtry = 10;
 
@@ -264,6 +268,8 @@
         LOGD("Unable to unload driver module \"%s\": %s\n",
              modname, strerror(errno));
     return ret;
+#endif
+	return 0;
 }
 
 int do_dhcp_request(int *ipaddr, int *gateway, int *mask,
