--- /psquare/developer/ANDROID/CYANOGENMOD7/frameworks/base/core/java/android/database/sqlite/SQLiteCursor.java
+++ /psquare/developer/ANDROID/CYANOGENMOD/frameworks/base/core/java/android/database/sqlite/SQLiteCursor.java
@@ -132,11 +132,15 @@
             // the cursor's state doesn't change
             while (true) {
                 mLock.lock();
+                if (mCursorState != mThreadState) {
+                    mLock.unlock();
+                    break;
+                }
                 try {
-                    if (mCursorState != mThreadState) {
-                        break;
-                    }
-
+//                    if (mCursorState != mThreadState) {
+//                        break;
+//                    }
+//
                     int count = mQuery.fillWindow(cw, mMaxRead, mCount);
                     // return -1 means not finished
                     if (count != 0) {
@@ -218,8 +222,10 @@
         mColumnNameMap = null;
         mQuery = query;
 
-        db.lock();
+//        db.lock();
         try {
+            db.lock();
+
             // Setup the list of columns
             int columnCount = mQuery.columnCountLocked();
             mColumns = new String[columnCount];
