--- current.xml.original	2012-02-01 21:21:42.000000000 +0100
+++ current.xml.psquare	2012-02-02 12:36:13.000000000 +0100
@@ -39307,6 +39307,17 @@
  visibility="public"
 >
 </field>
+<field name="ACTION_BRIGHTNESS_CHANGED"
+ type="java.lang.String"
+ transient="false"
+ volatile="false"
+ value="&quot;android.intent.action.BRIGHTNESS_CHANGED&quot;"
+ static="true"
+ final="true"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</field>
 <field name="ACTION_BUG_REPORT"
  type="java.lang.String"
  transient="false"
@@ -39934,6 +39945,17 @@
  visibility="public"
 >
 </field>
+<field name="ACTION_ROTATE_CHANGED"
+ type="java.lang.String"
+ transient="false"
+ volatile="false"
+ value="&quot;android.intent.action.ROTATION_SET&quot;"
+ static="true"
+ final="true"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</field>
 <field name="ACTION_RUN"
  type="java.lang.String"
  transient="false"
@@ -40099,6 +40121,17 @@
  visibility="public"
 >
 </field>
+<field name="ACTION_TVOUT_EVENT"
+ type="java.lang.String"
+ transient="false"
+ volatile="false"
+ value="&quot;android.intent.action.TVOUT_EVENT&quot;"
+ static="true"
+ final="true"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</field>
 <field name="ACTION_UID_REMOVED"
  type="java.lang.String"
  transient="false"
@@ -40736,6 +40769,39 @@
  visibility="public"
 >
 </field>
+<field name="EXTRA_TVOUT_STATE"
+ type="java.lang.String"
+ transient="false"
+ volatile="false"
+ value="&quot;android.intent.extra.TVOUT_STATE&quot;"
+ static="true"
+ final="true"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</field>
+<field name="EXTRA_TVOUT_STATE_OFF"
+ type="int"
+ transient="false"
+ volatile="false"
+ value="0"
+ static="true"
+ final="true"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</field>
+<field name="EXTRA_TVOUT_STATE_ON"
+ type="int"
+ transient="false"
+ volatile="false"
+ value="1"
+ static="true"
+ final="true"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</field>
 <field name="EXTRA_UID"
  type="java.lang.String"
  transient="false"
@@ -126522,6 +126588,28 @@
  visibility="public"
 >
 </method>
+<method name="getExternalStorage2Directory"
+ return="java.io.File"
+ abstract="false"
+ native="false"
+ synchronized="false"
+ static="true"
+ final="false"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</method>
+<method name="getExternalStorage2State"
+ return="java.lang.String"
+ abstract="false"
+ native="false"
+ synchronized="false"
+ static="true"
+ final="false"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</method>
 <method name="getExternalStorageDirectory"
  return="java.io.File"
  abstract="false"
@@ -126533,6 +126621,17 @@
  visibility="public"
 >
 </method>
+<method name="getExternalStorageLabel"
+ return="java.lang.String"
+ abstract="false"
+ native="false"
+ synchronized="false"
+ static="true"
+ final="false"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</method>
 <method name="getExternalStoragePublicDirectory"
  return="java.io.File"
  abstract="false"
@@ -126557,6 +126656,28 @@
  visibility="public"
 >
 </method>
+<method name="getInternalStorageDirectory"
+ return="java.io.File"
+ abstract="false"
+ native="false"
+ synchronized="false"
+ static="true"
+ final="false"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</method>
+<method name="getInternalStorageState"
+ return="java.lang.String"
+ abstract="false"
+ native="false"
+ synchronized="false"
+ static="true"
+ final="false"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</method>
 <method name="getRootDirectory"
  return="java.io.File"
  abstract="false"
@@ -126568,6 +126689,28 @@
  visibility="public"
 >
 </method>
+<method name="isExternalStorage2Removable"
+ return="boolean"
+ abstract="false"
+ native="false"
+ synchronized="false"
+ static="true"
+ final="false"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</method>
+<method name="isExternalStorageBeSdcard"
+ return="boolean"
+ abstract="false"
+ native="false"
+ synchronized="false"
+ static="true"
+ final="false"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</method>
 <method name="isExternalStorageRemovable"
  return="boolean"
  abstract="false"
@@ -148285,6 +148428,28 @@
  visibility="public"
 >
 </field>
+<field name="FORCE_ROTATION"
+ type="java.lang.String"
+ transient="false"
+ volatile="false"
+ value="&quot;force_rotation&quot;"
+ static="true"
+ final="true"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</field>
+<field name="FORCE_ROTATION_DIRECTION"
+ type="java.lang.String"
+ transient="false"
+ volatile="false"
+ value="&quot;force_rotation_direction&quot;"
+ static="true"
+ final="true"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</field>
 <field name="HAPTIC_FEEDBACK_ENABLED"
  type="java.lang.String"
  transient="false"
@@ -182813,6 +182978,17 @@
  visibility="public"
 >
 </field>
+<field name="KEYCODE_MEDIA_AUDIO"
+ type="int"
+ transient="false"
+ volatile="false"
+ value="118"
+ static="true"
+ final="true"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</field>
 <field name="KEYCODE_MEDIA_FAST_FORWARD"
  type="int"
  transient="false"
@@ -182824,6 +183000,17 @@
  visibility="public"
 >
 </field>
+<field name="KEYCODE_MEDIA_INFO"
+ type="int"
+ transient="false"
+ volatile="false"
+ value="115"
+ static="true"
+ final="true"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</field>
 <field name="KEYCODE_MEDIA_NEXT"
  type="int"
  transient="false"
@@ -182857,6 +183044,28 @@
  visibility="public"
 >
 </field>
+<field name="KEYCODE_MEDIA_RATIO"
+ type="int"
+ transient="false"
+ volatile="false"
+ value="116"
+ static="true"
+ final="true"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</field>
+<field name="KEYCODE_MEDIA_REPEAT"
+ type="int"
+ transient="false"
+ volatile="false"
+ value="114"
+ static="true"
+ final="true"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</field>
 <field name="KEYCODE_MEDIA_REWIND"
  type="int"
  transient="false"
@@ -182879,6 +183088,17 @@
  visibility="public"
 >
 </field>
+<field name="KEYCODE_MEDIA_SUB_T"
+ type="int"
+ transient="false"
+ volatile="false"
+ value="117"
+ static="true"
+ final="true"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</field>
 <field name="KEYCODE_MENU"
  type="int"
  transient="false"
@@ -183231,6 +183451,17 @@
  visibility="public"
 >
 </field>
+<field name="KEYCODE_TV_SYSTEM"
+ type="int"
+ transient="false"
+ volatile="false"
+ value="113"
+ static="true"
+ final="true"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</field>
 <field name="KEYCODE_U"
  type="int"
  transient="false"
@@ -183330,6 +183561,28 @@
  visibility="public"
 >
 </field>
+<field name="KEYCODE_ZOOM_IN"
+ type="int"
+ transient="false"
+ volatile="false"
+ value="111"
+ static="true"
+ final="true"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</field>
+<field name="KEYCODE_ZOOM_OUT"
+ type="int"
+ transient="false"
+ volatile="false"
+ value="112"
+ static="true"
+ final="true"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</field>
 <field name="MAX_KEYCODE"
  type="int"
  transient="false"
@@ -197691,6 +197944,17 @@
  visibility="public"
 >
 </field>
+<field name="TYPE_APPLICATION_KEYGUARD"
+ type="int"
+ transient="false"
+ volatile="false"
+ value="98"
+ static="true"
+ final="true"
+ deprecated="not deprecated"
+ visibility="public"
+>
+</field>
 <field name="TYPE_APPLICATION_MEDIA"
  type="int"
  transient="false"
@@ -208873,6 +209137,19 @@
 <parameter name="certificate" type="android.net.http.SslCertificate">
 </parameter>
 </method>
+<method name="setDoubleTapForZoom"
+ return="void"
+ abstract="false"
+ native="false"
+ synchronized="false"
+ static="false"
+ final="false"
+ deprecated="not deprecated"
+ visibility="public"
+>
+<parameter name="bool" type="boolean">
+</parameter>
+</method>
 <method name="setDownloadListener"
  return="void"
  abstract="false"
