--- CameraService.cpp.original	2012-02-09 05:39:01.000000000 +0100
+++ CameraService.cpp.psquare	2012-02-09 06:21:43.000000000 +0100
@@ -41,6 +41,8 @@
 
 #include "CameraService.h"
 
+static void convert_rgb16_to_yuv420sp(uint8_t *rgb, uint8_t *yuv, int width, int height);
+
 #if defined(USE_OVERLAY_FORMAT_YCbCr_420_SP) || defined(USE_OVERLAY_FORMAT_YCrCb_420_SP) || defined(CAF_CAMERA_GB_REL)
 #include "gralloc_priv.h"
 #endif
@@ -1577,7 +1579,7 @@
 #endif
 
         ISurface::BufferHeap buffers(w, h, w, h,
-            mPixelFormat, mOrientation, 0,
+            HAL_PIXEL_FORMAT_YCrCb_420_SP, mOrientation, 0,
             mHardware->getRawHeap());
 
         mSurface->registerBuffers(buffers);
